### PR TITLE
feat: display all address in the ckb2021 format

### PR DESCRIPTION
**Pull request by homura**
This PR 

- modifies the app to display all addresses in the CKB2021 format
- refactors the code to migrate the deprecated `cx` API to the latest

### CODE REVIEW

File: src/segwit_addr.c [hunk 1]
```diff
@@ -54,7 +54,7 @@ int bech32_encode(char *const output, const size_t out_len, const char *const hr
             chk = bech32_polymod_step(chk) ^ (hrp[i] >> 5);
             ++i;
         }
-        if (i + 7 + data_len > 108)
+        if (i + 7 + data_len > 1023)
```
Comment (cedelavergne-ledger) [inline src/segwit_addr.c:57]: Could be interesting to have a comment explaining this hardcoded value...

File: src/apdu_pubkey.c [hunk 2]
```diff
@@ -53,19 +53,10 @@ static void bip32_path_to_string(char *const out, size_t const out_size, apdu_pu
...
-    bool is_bech32m = 0;
-    if (payload->full_version.address_format_type == ADDRESS_FORMAT_TYPE_FULL_VERSION) {
-        payload_len = sizeof(payload->full_version);
-        is_bech32m = 1;
-    } else if (payload->short_version.address_format_type == ADDRESS_FORMAT_TYPE_SHORT) {
-        payload_len = sizeof(payload->short_version);
-    } else {
-        payload_len = sizeof(payload->code_hash_data_or_type);
-    }
+    bool is_bech32m = 1;
```
Comment (cedelavergne-ledger) [inline src/apdu_pubkey.c:59]: Why do you keep a boolean variable with only hardcoded `true` value?

**Comment by cedelavergne-ledger** [comment]
Could be interesting/better to use `CX_CHECK` and `CX_ASSERT` instead of `CX_THROW`, allowing a better retrun codes management...

File: src/apdu_pubkey.c (hunk 2 above)
Comment (homura) [reply src/apdu_pubkey.c:59]: This is due to the deprecation of the bech32 (short address format), which now only uses [bech32m](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0021-ckb-address-format/0021-ckb-address-format.md#full-payload-format) to encode a script to an address. All test cases under `/tests` were encoded with bech32m.

File: src/segwit_addr.c (hunk 1 above)
Comment (homura) [reply src/segwit_addr.c:57]: Sure, I'll add a comment for the magic number

File: src/apdu_pubkey.c (hunk 2 above)
Comment (cedelavergne-ledger) [reply src/apdu_pubkey.c:59]: Ok, but this variable could then be removed, and a `1` or `true` value can be passed to the function `bech32_encode`

**Comment by Keith-CY** [comment]
> Could be interesting/better to use `CX_CHECK` and `CX_ASSERT` instead of `CX_THROW`, allowing a better retrun codes management...

Hi @cedelavergne-ledger
Code has been updated, please have a review

**Comment by cedelavergne-ledger** [comment]
> > Could be interesting/better to use `CX_CHECK` and `CX_ASSERT` instead of `CX_THROW`, allowing a better retrun codes management...
> 
> Hi @cedelavergne-ledger Code has been updated, please have a review

:thinking: 
Code has not changed recently, and I can still see the `CX_THROW`... Do I miss something, or do you forget to push your modifications?

**Review by cedelavergne-ledger — APPROVED** [review]

**Pull request merged**